### PR TITLE
Adds proxy support

### PR DESCRIPTION
**Pull request by godinezj**

### CODE REVIEW

**Comment by nickgrealy** [comment]
Hmm... Jenkins is complaining about this:

https://jenkins.ci.cloudbees.com/job/plugins/job/aws-sqs-plugin/2/console

```Shell
[INFO] --- findbugs-maven-plugin:3.0.3:check (findbugs) @ aws-sqs ---

[INFO] BugInstance size is 1
[INFO] Error size is 0
[INFO] Total bugs: 1
[INFO] Possible null pointer dereference in io.relution.jenkins.awssqs.factories.SQSFactoryImpl.getClientConfiguration(SQSQueue) due to return value of called method [io.relution.jenkins.awssqs.factories.SQSFactoryImpl, io.relution.jenkins.awssqs.factories.SQSFactoryImpl] Dereferenced at SQSFactoryImpl.java:[line 102]Known null at SQSFactoryImpl.java:[line 102] NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE
[INFO] 

To see bug detail using the Findbugs GUI, use the following command "mvn findbugs:gui"
```

Not sure if it's a new or existing issue (looks like Jenkins/CloudBees have a new build process?). I'll take a quick look...

**Comment by nickgrealy** [comment]
Hi @godinezj ,

The build issue seems to be introduced with your code. I tested with `master` vs your new branch.

To replicate, you'll need to run:

```Shell
mvn -V clean install -e -U -Dignore.random.failures -Dconcurrency=2
```

Other than that, the pull looks fine. Get the build passing, and I'll approve the pull request. Thanks!

Nick

File: src/main/java/io/relution/jenkins/awssqs/factories/SQSFactoryImpl.java [hunk 1]
```diff
@@ -94,8 +99,22 @@ public SQSChannel createChannel(final io.relution.jenkins.awssqs.interfaces.SQSQ
     private ClientConfiguration getClientConfiguration(final io.relution.jenkins.awssqs.interfaces.SQSQueue queue) {
         final ClientConfiguration config = new ClientConfiguration();
 
-        // TODO Add support for proxy
-
+        ProxyConfiguration proxyConfig = Jenkins.getInstance().proxy;
```
Comment (nickgrealy) [inline src/main/java/io/relution/jenkins/awssqs/factories/SQSFactoryImpl.java]: This is the line Jenkins Build is complaining about. I'd try adding a null check on the `Jenkins.getInstance()` method.

Comment (godinezj) [inline src/main/java/io/relution/jenkins/awssqs/factories/SQSFactoryImpl.java]: ok, please verify.

**Pull request merged**